### PR TITLE
Only flag indexer ready after commit

### PR DESCRIPTION
**Pull request by rorygraves**
As discussed, adding the tests highlighted a race condition which I believe this fixes.
@fommil  want to take a gander  - also see inline comment about results.

### CODE REVIEW

File: src/test/scala/org/ensime/intg/BasicWorkflow.scala [hunk 1]
```diff
@@ -66,6 +66,33 @@ class BasicWorkflow extends FunSpec with Matchers {
...
+        // public symbol search - java.io.File
+
+        val javaSearchSymbol = project.rpcPublicSymbolSearch(List("java", "io", "File"), 2)
+        javaSearchSymbol match {
+          case SymbolSearchResults(List(
+            TypeSearchResult("java.awt.image.ImageObserver", "ImageObserver", 'class,
+              Some(_)),
+            TypeSearchResult("java.io.File", "File", 'class,
+              Some(_)),
+            TypeSearchResult("org.ensime.util.FileUtilSpec", "FileUtilSpec", 'class, None))) =>
```
Comment (rorygraves) [inline src/test/scala/org/ensime/intg/BasicWorkflow.scala]: @fommil These indexer results seem wrong - two items:
1. It asks for 2 items and gets 3
2. Search for "java" "io" "File" returns j.a.i.ImageObserver and FileUtilSpec

I accept I may not understand the protocol spec but this feels a bit random.

File: src/main/scala/org/ensime/core/Indexer.scala [hunk 2]
```diff
@@ -53,11 +51,6 @@ class Indexer(
   }
 
   override def receive = {
-    case SourceFileCandidatesReq(enclosingPackage, classNamePrefix) =>
```
Comment (rorygraves) [inline src/main/scala/org/ensime/core/Indexer.scala]: This was dead and uncallable code so I ripped it out

File: src/test/scala/org/ensime/intg/BasicWorkflow.scala (hunk 1 above)
Comment (fommil) [inline src/test/scala/org/ensime/intg/BasicWorkflow.scala]: 1. that looks like a bug. The max number must not be going through to lucene. Also see #637 
2. this is actually reasonable: "io" will hit the `ImageObserver` via CamelCase... but `java.io.File` should be a much stronger hit.

**Comment by fommil** [comment]
good catch! I was too close to it to see it.

File: src/test/scala/org/ensime/intg/BasicWorkflow.scala (hunk 1 above)
Comment (rorygraves) [inline src/test/scala/org/ensime/intg/BasicWorkflow.scala]: I assumed on first reading that it was an expression like j.i.File where the subparts could be camel or partials but all would have to match.
I will raise a new ticket for the count break (I looked, it appears to pass all the way to Lucene)

**Pull request merged**